src/ansiblelint/rules/name.py

```diff
@@ -35,14 +60,23 @@ def matchtask(
                 tag="name[missing]",
                 filename=file,
             )
+        return (
+            self._check_name(name, filename=file, linenumber=task[LINE_NUMBER_KEY])
+            or False
+        )
+
+    # pylint: disable=useless-return
+    def _check_name(
+        self, name: str, filename: "Optional[Lintable]", linenumber: int
+    ) -> "Optional[MatchError]":
         if not name[0].isupper():
-            return self.create_matcherror(
+            self.create_matcherror(
```

Comment:
```suggestion
            return self.create_matcherror(
```

You're not returning the MatchError here. If you fix this, does the pylint error go away?